salt/_modules/metalk8s.py

```diff
@@ -82,7 +82,7 @@ def _execute_etcd_command(pod_name, cmd):
     api = kubernetes.client.CoreV1Api(api_client=client)
```

Comment:
This whole procedure should no longer be required, no?

Reply:
It is still used for now. This is used by the upgrade orchestrator in order to check the etcd heath. The issue is that `etcd3` does not provide a method to check the cluster health. We have the status method (https://python-etcd3.readthedocs.io/en/latest/_modules/etcd3/client.html#Etcd3Client.status) but it seems, that it does not return the cluster health

Reply:
That's because you can't "ask" the cluster about the health of the cluster (you can't do that in a quorum-based distributed system...). All you can do is have a client request/figure out the status of cluster nodes, and derive a cluster health from it.

Reply:
https://github.com/etcd-io/etcd/blob/master/etcdctl/ctlv2/command/cluster_health.go

Reply:
Ok, will fix